qframe.go

```diff
@@ -1106,8 +1106,26 @@ func (qf QFrame) ToCSV(writer io.Writer, confFuncs ...csv.ToConfigFunc) error {
 		return qerrors.Propagate("ToCSV", qf.Err)
 	}
 
-	row := make([]string, 0, len(qf.columns))
-	for _, s := range qf.columns {
+	var iterCols []namedColumn
+	if conf.Columns != nil {
+		if len(conf.Columns) != len(qf.columns) {
+			return qerrors.New("ToCSV", fmt.Sprintf("wrong number of columns: expected: %d", len(qf.columns)))
+		}
+		iterCols = make([]namedColumn, 0, len(qf.columns))
```

Comment:
Just some nitpicking and suggestions for how to make the code a bit shorter and more idiomatic:
```go
		iterCols = make([]namedColumn, len(qf.columns))    // Pre-set length
		for i := range conf.Columns {    // Use range expression
			cName := conf.Columns[i]  // camelCase name
			if col, ok := qf.columnsByName[cName]; !ok {
				return qerrors.New("ToCSV", fmt.Sprintf("%s: column does not exist in QFrame", cName))
			}
                        // Skip the else-clause since there is a return in the if-clause.

                        // Set element using indexing rather than append (since we now have a length set)
			iterCols[i] = col
		}
```

Reply:
Nice, thank you @tobgu 
Made all recommended changes, except the last one: when `col` is defined inline in the `if`, then it needs a `else` branch to be used, or the compiler will reject the code. See minimal example here: https://go.dev/play/p/2bowv-oHDl5